src/server/network/network.ts

```diff
@@ -0,0 +1,93 @@
+import dgram = require('dgram');
+
+export type ClientListener = (event: string, ...args: any[]) => void;
+
+type TeamData = {
+  server: dgram.Socket;
+  name: string;
+  address: string;
+  port: number;
+};
+
+export class UDPServer {
+  private redTeam: TeamData;
+  private blueTeam: TeamData;
+  private clients: Map<string, ClientListener>;
+
+  constructor(redTeamAddress: string, blueTeamAddress: string) {
+    this.clients = new Map();
+
+    this.redTeam = {
+      server: dgram.createSocket({ type: 'udp4', reuseAddr: true }),
+      name: 'Red Team',
+      address: redTeamAddress,
+      port: 9917,
+    };
+    this.blueTeam = {
+      server: dgram.createSocket({ type: 'udp4', reuseAddr: true }),
+      name: 'Blue Team',
+      address: blueTeamAddress,
+      port: 9917,
+    };
+
+    this.initTeam(this.redTeam);
+    this.initTeam(this.blueTeam);
```

Comment:
This is gross if you are going to basically write a function that's a constructor just make a new Class.
Or more likely do what I said earlier and make one of these objects be for only a single UDPServer